specs/AdaptiveCardsXamarinSpec.md

```diff
@@ -0,0 +1,188 @@
+
+### What is a bindings library
+
+A bindings library is an assembly that wraps the native library with C# wrappers so native code can be invoked via C# calls. Xamarin.Android implements bindings by using Managed Callable Wrappers (MCW) which are a JNI bridge that are used when managed code needs to invoke Java code. In the case of iOS, to bind a library an API definition file is needed; this is merely a C# source file that contains C# interfaces that have been annotated with a handful of attributes that help drive the binding. The API definition is merely the contract that will be used to generate the API.
+
+## Architecture
+
+As can be seen in the diagram below, the Android and iOS bindings libraries make use of the Java and Objective-C code that is already projected from the shared C++ code. The bindings also make use of the native APIs for both platforms yielding a C# library for every platform that can be consumed by any Xamarin platform.
+
+[image](assets/XamarinArchitecture.png)
+
+Something to consider is that Native Xamarin is designed to share the logic and services inside an application but the UI is independent for every platform which allows us to have three different object models.  
+
+## Rendering
+
+### UWP
+
+For Xamarin rendering UWP doesn't require any changes as the rendering and API's are done in C#. The current renderer we provide must only be referenced as is or included as a nuget package.
+
+### Android
+
+Android bindings are mostly done in the Metadata.xml file which defines the contract between C# and Java. This file can contain renames for packages, change of return types and parameter types and names as well as removing classes or including extra classes.
+
+``` xml
+<metadata>
+  <attr path="/api/package[@name='io.adaptivecards']" name="managedName">AdaptiveCards.Rendering.Xamarin.Android</attr>
+
+  <!-- Updates Get methods to return java.lang.Object instead of the actual type -->
+  <attr path="/api/package[@name='io.adaptivecards.objectmodel']/class[@name='BaseActionElementVector']/ method[@name='get']" name="managedReturn">Java.Lang.Object</attr>
+
+  <!-- Updates DoInBackground method's parameter and return types -->
+  <attr path="/api/package[@name='io.adaptivecards.renderer']/class[@name='BackgroundImageLoaderAsync']/ method[@name='doInBackground']" name="managedReturn">Java.Lang.Object</attr>
+
+  <add-node path="/api/package[@name='io.adaptivecards.renderer']">
+    <class abstract="false" deprecated="not deprecated" final="false" name="MediaLoaderAsync" static="true" visibility="public" extends="java.lang.Object"/>
+  </add-node>
+</metadata>
+```
+
+#### APIs to be included in Xamarin
+
+| API Name | Android API |  Priority |
+| --- | --- | --- |
+| Parsing | ```ParseResult Deserialize(JsonValue, String, ParseContext)``` | P0 |
+| Parsing | ```ParseResult DeserializeFromString(String, String)``` | P0 |
+| Parsing | ```ParseResult DeserializeFromString(String, String, ParseContext)``` | P0 |
```

Comment:
[Team Spec Review] same comment as below. Order by priority making the first column as the priority